lang/java/avro/src/main/java/org/apache/avro/specific/SpecificData.java

```diff
@@ -149,7 +149,7 @@ public static SpecificData get() {
   }
 
   /**
-   * For RECORD type schemas, this method returns the SpecificData instance of the
+   * For RECORD and UNION type schemas, this method returns the SpecificData instance of the
```

Comment:
Is this done to also use logical types on record values? 
If so, would it make sense to support all container types? So record, union, array and map?

Reply:
happens only with records under union. Added test cases for the same